src/test/java/itest/RulesPostFormIT.java

```diff
@@ -0,0 +1,149 @@
+ * SOFTWARE.
+ */
+package itest;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import io.cryostat.net.web.http.HttpMimeType;
+
+import io.vertx.core.MultiMap;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.json.JsonObject;
+import itest.bases.StandardSelfTest;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RulesPostFormIT extends StandardSelfTest {
+
+    static MultiMap testRule;
+
+    @BeforeAll
+    static void setup() throws Exception {
+        testRule = MultiMap.caseInsensitiveMultiMap();
+        testRule.add("name", "Test Rule");
+        testRule.add("targetAlias", "es.andrewazor.demo.Main");
+        testRule.add("description", "AutoRulesIT automated rule");
+        testRule.add("eventSpecifier", "template=Continuous,type=TARGET");
+    }
+
+    @Test
+    void testAddRuleThrowsWhenFormAttributesNull() throws Exception {
+        CompletableFuture<JsonObject> response = new CompletableFuture<>();
+
+        webClient
+                .post("/api/v2/rules")
+                .putHeader(HttpHeaders.CONTENT_TYPE.toString(), HttpMimeType.MULTIPART_FORM.mime())
+                .sendForm(
+                        null,
+                        ar -> {
+                            if (assertRequestStatus(ar, response)) {
+                                MatcherAssert.assertThat(
+                                        ar.result().statusCode(), Matchers.equalTo(400));
+                            }
+                        });
+    }
+
+    @Test
+    void testAddRuleThrowsWhenRuleNameAlreadyExists() throws Exception {
+        CompletableFuture<JsonObject> postResponse = new CompletableFuture<>();
+        CompletableFuture<JsonObject> duplicatePostResponse = new CompletableFuture<>();
+        CompletableFuture<JsonObject> deleteResponse = new CompletableFuture<>();
+
+        webClient
+                .post("/api/v2/rules")
+                .sendForm(
+                        testRule,
+                        ar -> {
+                            if (assertRequestStatus(ar, postResponse)) {
+                                MatcherAssert.assertThat(
+                                        ar.result().statusCode(), Matchers.equalTo(201));
+                                postResponse.complete(ar.result().bodyAsJsonObject());
+                            }
+                        });
+        postResponse.get();
+
+        webClient
+                .post("/api/v2/rules")
+                .sendForm(
+                        testRule,
+                        ar -> {
+                            if (assertRequestStatus(ar, duplicatePostResponse)) {
+                                MatcherAssert.assertThat(
+                                        ar.result().statusCode(), Matchers.equalTo(409));
```

Comment:
I think this works but not for the reason you think it does - check the implementation of `assertRequestStatus`.